### ISSUE TITLE
What kind of auction simulation do we want to build?

### ISSUE THREAD

**Issue by davidrpugh** [EconomicSL/auctions-simulation-example#1]
@ibillett What kind of auction simulation should we develop?  Perhaps ask your friends at work for ideas?

**Comment by ghost** [comment]
So the key feature of these models is that they can demonstrate dynamics that can't be captured in structural macro models. This should be our starting point, and will capture people's attention.

A demo that has consequences for promoting monetary and financial stability will always get 👍  from central bankers. Some kind of endogenous liquidity shortage / crisis, would be really interesting.

Just to be 100% clear, by auction you mean simulating the buying and selling of an asset between different agents? I'm sure it has a precise economic definition that I am ignorant to.

I'll pose the question internally and see if there are any better ideas, though I'm sure they will follow the above logic.

**Comment by davidrpugh** [comment]
> So the key feature of these models is that they can demonstrate dynamics that can't be captured in structural macro models. This should be our starting point, and will capture people's attention.

Agreed with the caveat that the auctions library is only one of several libraries that would need to be combined in order to produce a macroeconomic simulation.  We should be able to generate interesting dynamics for a single market however.

> A demo that has consequences for promoting monetary and financial stability will always get 👍 from central bankers. Some kind of endogenous liquidity shortage / crisis, would be really interesting.

It would be good to do something policy relevant.  A former colleague of mine at Oxford, Daniel Fricke, wrote a paper on the [optimal speed of financial markets](https://www.sec.gov/files/dera-wp-optimal-speed.pdf). We could do something similar: set up a periodic double auction simulation where one of the parameters is the time interval between clearings.  We could then vary the clearing interval to see what impact it has on price volatility, or something.

> Just to be 100% clear, by auction you mean simulating the buying and selling of an asset between different agents? I'm sure it has a precise economic definition that I am ignorant to.

In this case yes.  However, the ESL auctions library is for general purpose auction simulation for any tradable good.
